lib/puppet/provider/service/systemd.rb

```diff
@@ -127,10 +127,10 @@ def debian_enabled?
   # This function is called only on start & restart unit options.
   # Reference: (PUP-3483) Systemd provider doesn't scan for changed units
   def daemon_reload?
-    cmd = [command(:systemctl), 'show', @resource[:name], '--property=NeedDaemonReload']
+    cmd = [command(:systemctl), 'show', '--', @resource[:name], '--property=NeedDaemonReload']
     daemon_reload = execute(cmd, :failonfail => false).strip.split('=').last
     if daemon_reload == 'yes'
-      daemon_reload_cmd = [command(:systemctl), 'daemon-reload']
+      daemon_reload_cmd = [command(:systemctl), '--', 'daemon-reload']
       execute(daemon_reload_cmd, :failonfail => false)
     end
   end
```

Comment:
`daemon_reload?` is from 6.4.x, so the `--` is in addition to the changes of 5.5.x